### PR TITLE
[Docs] Generate man pages

### LINKED ISSUE #11399: Writing Manpage
## Description

It is convenient to have manpage for viewing Gitea docs (especially [Command Line](https://docs.gitea.io/en-us/command-line/) and [Config Cheat Sheet](https://docs.gitea.io/en-us/config-cheat-sheet/)) locally.

The manpage will be written using go-md2man as suggested by @lafriks.

The workflow (as described in [Hacking doc](https://docs.gitea.io/en-us/hacking-on-gitea/) should be modified to include that any changes to config cheat sheet and/or command line options list docs above, the corresponding changes should also be made onto manpage.

However, how should the manpage be distributed along with Gitea? Currently [dl page](https://dl.gitea.io/gitea) distribute Gitea as single binary and xz-compressed of it. So I had two options:

1. Distribute manpage separately, as xz-compressed, or
2. For compressed archive of Gitea binary, throw manpage into it.

### PR DESCRIPTION
**Pull request by 6543**
close  #11476
fix #11399

### CODE REVIEW

**Comment by 6543** [comment]
@bagasme would you like to have a look at it

**Comment by jolheiser** [comment]
This should probably leverage the existing docs command with the [man flag](https://github.com/go-gitea/gitea/blob/42354dfe45fa0cabb59674b896c44a55a56cf163/cmd/docs.go#L22-L25).

**Comment by lunny** [comment]
Is this ready to review?

**Comment by 6543** [comment]
@lunny sure - for the whole time

**Review by jolheiser — COMMENTED** [review]
Consider leverage the existing docs command with the [man flag](https://github.com/go-gitea/gitea/blob/42354dfe45fa0cabb59674b896c44a55a56cf163/cmd/docs.go#L22-L25).

File: Makefile [hunk 1]
```diff
@@ -700,6 +702,20 @@ generate-images:
 	npm install --no-save --no-package-lock fabric imagemin-zopfli
 	node build/generate-images.js $(TAGS)
 
+.PHONY: generate-manpage
+generate-manpage:
+	@[ -f gitea ] || make backend
+	@mkdir -p man/man1/ man/man5
+	@./gitea docs --man > man/man1/gitea.1
```
Comment (6543) [inline Makefile:710]: in usage ...

Comment (jolheiser) [reply Makefile:710]: Yep, my bad. I saw the below usage of `go-md2man` and my brain ignored the rest. 😓

File: Makefile [hunk 2]
```diff
@@ -700,6 +702,20 @@ generate-images:
...
+	@gzip -9 man/man1/gitea.1 && echo man/man1/gitea.1.gz created
+	@#TODO A smal script witch format config-cheat-sheet.en-us.md nicely to suit as man page
+	@tail -n +36 docs/content/doc/advanced/config-cheat-sheet.en-us.md > man/man5/gitea.app.ini.5.md
```
Comment (jolheiser) [inline Makefile]: I don't know that I like this part. The config cheat sheet changes [fairly often](https://github.com/go-gitea/gitea/commits/master/docs/content/doc/advanced/config-cheat-sheet.en-us.md) and an arbitrary `+36` seems prone to breaking.  
To be fair, the changes are usually in the config parts, which are after line 36, however any changes to the beginning paragraphs would potentially break this.

As a first PR maybe the docs man pages are enough? Can try to get something more precise to add the config later?

**Comment by 6543** [comment]
@jolheiser removed it and just leave a TODO back there ...

**Review by jolheiser — APPROVED** [review]

**Review by zeripath — APPROVED** [review]

**Pull request merged**